framework/src/org/checkerframework/common/value/util/Range.java

```diff
@@ -853,26 +886,39 @@ private boolean isWithinInteger() {
      * <p>If the BigInteger range is too wide, i.e., wider than the full range of the Long class,
      * return EVERYTHING.
      *
-     * <p>If the BigInteger bounds are out of the Long type scope, convert the bounds to Long type
-     * in accordance with Java overflow rules, e.g., Long.MAX_VALUE + 1 is converted to
-     * Long.MIN_VALUE.
+     * <p>If one of the BigInteger bounds is out of Long's range and this range does not ignore
+     * overflow, convert the bounds to Long type in accordance with Java overflow rules, e.g.,
+     * Long.MAX_VALUE + 1 is converted to Long.MIN_VALUE.
+     *
+     * <p>If one of the BigInteger bounds is out of Long's range and this range ignores overflow,
+     * convert the bound that is outside Long's range to max/min value of a Long.
      *
      * @param bigFrom the lower bound of the BigInteger range
      * @param bigTo the upper bound of the BigInteger range
      * @return a range with Long type bounds converted from the BigInteger range
      */
-    private static Range bigRangeToLongRange(BigInteger bigFrom, BigInteger bigTo) {
+    private Range bigRangeToLongRange(BigInteger bigFrom, BigInteger bigTo) {
         BigInteger numValues = bigTo.subtract(bigFrom).add(BigInteger.ONE);
-        if (numValues.compareTo(longWidth) == 1) {
-            return EVERYTHING;
+        long resultFrom;
+        long resultTo;
+        if (IGNORE_OVERFLOW) {
+            BigInteger longMin = BigInteger.valueOf(Long.MIN_VALUE);
+            resultFrom = bigFrom.max(longMin).longValue();
+            BigInteger longMax = BigInteger.valueOf(Long.MAX_VALUE);
+            resultTo = bigTo.min(longMax).longValue();
         } else {
-            long resultFrom = bigFrom.longValue();
-            long resultTo = bigTo.longValue();
-            if (resultFrom <= resultTo) {
-                return new Range(resultFrom, resultTo);
-            } else {
+            if (numValues.compareTo(longWidth) == 1) {
```

Comment:
`compareTo` is allowed to return any positive value.  So, you should compare its result to 0, but don't assume it returns 1.